File: day04_1.php
<?php
  class PassphraseCounter
  {
    public function count_correct($input)
    {
      $count = 0;
      foreach (explode(PHP_EOL, $input) as $line) {
        $words = explode(' ', trim($line));
        if (array_unique($words) == $words) $count++;
      }
      return $count;
    }
  }

  $passphrase_counter = new PassphraseCounter;

  $input = <<<EOT
... paste input here ...
EOT;

  echo $passphrase_counter->count_correct($input);

File: 2023/day04_1.rb
f = __FILE__.sub(/\d\..*/, 'input')

p File.foreach(f).sum { |line|
  _, win, own = line.split(/[:|]/).map { _1.scan(/\d+/) }
  (own & win).count.then { |n| n > 0 ? 2 ** (n - 1) : 0 }
}
